### PR TITLE
Upgrading k8s lib

### PR DESCRIPTION
**Pull request by wilmer05**
We need to update the kubernetes library to newer versions. 

Containerd gives problems when looking for container image names with the kubernetes client in the current version.

PAASTA-18345

### CODE REVIEW

File: paasta_tools/kubernetes_tools.py [hunk 1]
```diff
@@ -610,9 +608,9 @@ def __init__(
         # We need to support apiextensions /v1 and /v1beta1 in order
         # to make our upgrade to k8s 1.22 smooth, otherwise
         # updating the CRDs make this script fail
-        self.apiextensions_v1_beta1 = kube_client.ApiextensionsV1beta1Api(
-            self.api_client
-        )
+        # self.apiextensions_v1_beta1 = kube_client.ApiextensionsV1beta1Api(
```
Comment (ilkinmammadzada) [inline paasta_tools/kubernetes_tools.py:611]: was it dead code?

Comment (wilmer05) [reply paasta_tools/kubernetes_tools.py]: I removed it now

Comment (nemacysts) [reply paasta_tools/kubernetes_tools.py]: if we need this code again we can grab it from git history :)

File: paasta_tools/kubernetes_tools.py [hunk 2]
```diff
@@ -4155,12 +4153,8 @@ def mode_to_int(mode: Optional[Union[str, int]]) -> Optional[int]:
...
 def update_crds(
     kube_client: KubeClient,
-    desired_crds: Collection[
-        Union[V1CustomResourceDefinition, V1beta1CustomResourceDefinition]
-    ],
-    existing_crds: Union[
-        V1CustomResourceDefinitionList, V1beta1CustomResourceDefinitionList
-    ],
+    desired_crds: Collection[Union[V1CustomResourceDefinition]],
+    existing_crds: Union[V1CustomResourceDefinitionList],
```
Comment (nemacysts) [inline paasta_tools/kubernetes_tools.py:4151]:
```suggestion
    desired_crds: Collection[V1CustomResourceDefinition],
    existing_crds: V1CustomResourceDefinitionList,
```
we don't need a union anymore since we're using a single type now :)

File: paasta_tools/setup_kubernetes_cr.py [hunk 3]
```diff
@@ -175,7 +174,7 @@ def setup_all_custom_resources(
             ).items
         except ApiException:
             log.debug(
-                "Listing CRDs with apiextensions/v1 not supported on this cluster, falling back to v1beta1"
+                "Listing CRDs with apiextensions/v1 not supported on this cluster"
```
Comment (nemacysts) [inline paasta_tools/setup_kubernetes_cr.py:177]: imo: i think we'd want to revert the changes to this file where we added the for loop rather than make this a for loop over a single version

Comment (nemacysts) [reply paasta_tools/setup_kubernetes_cr.py:177]: (it would make the code more straightforward and while we may need to handle new apiextension versions in the future, imo it'd be better to only temporarily have this sort of multi-version code)

**Review by nemacysts — APPROVED** [review]
